nvflare/private/fed/server/server_runner.py

```diff
@@ -122,12 +123,15 @@ def _execute_run(self):
             wf = self.config.workflows[self.current_wf_index]
             try:
                 with self.engine.new_context() as fl_ctx:
-                    self.log_info(fl_ctx, "starting workflow {} ({}) ...".format(wf.id, type(wf.responder)))
+                    self.log_info(fl_ctx, "starting workflow {} ({}) ...".format(wf.id, type(wf.controller)))
 
                     fl_ctx.set_prop(FLContextKey.WORKFLOW, wf.id, sticky=True)
-                    wf.responder.initialize_run(fl_ctx)
 
-                    self.log_info(fl_ctx, "Workflow {} ({}) started".format(wf.id, type(wf.responder)))
+                    wf.controller.set_communicator(WFCommServer(), fl_ctx)
```

Comment:
instead of creating `WFCommServer()` here, I think it is better to do set_communicator part in the "Workflow" object construction? Since that object is only on the server side?

Reply:
we can do the `set_engine/initialize` here with fl_ctx.